csharp/src/Ice/Communicator.cs

```diff
@@ -51,6 +54,71 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
         /// </returns>
         public ILogger Logger { get; internal set; }
 
+        /// <summary>Each time you send a request without an explicit context parameter, Ice sends automatically the
+        /// per-thread CurrentContext combined with the proxy's context.</summary>
+        public Context CurrentContext
+        {
+            get
+            {
+                try
+                {
+                    if (_currentContext.IsValueCreated)
+                    {
+                        return _currentContext.Value;
+                    }
+                    else
+                    {
+                        _currentContext.Value = new Context();
+                        return _currentContext.Value;
+                    }
+                }
+                catch (ObjectDisposedException ex)
```

Comment:
Is this really useful can just the caller handle ObjectDisposedException

Reply:
I think it's useful and I added this because a test caught it. The use-case is not a direct call to CurrentContext by the application, but when the application calls an operation on a proxy with a destroyed communicator: `CurrentContext` (used as part of the request building) is actually the first call to throw an exception.